src/target/lpc546xx.c

```diff
@@ -98,55 +136,23 @@ bool lpc546xx_probe(target_s *t)
 {
 	const uint32_t chipid = target_mem_read32(t, LPC546XX_CHIPID);
 	uint32_t flash_size = 0;
+	uint32_t sram123_size = 0;
 
-	switch (chipid) {
-	case 0x7f954605U:
-		t->driver = "LPC54605J256";
-		flash_size = 0x40000;
-		break;
-	case 0x7f954606U:
-		t->driver = "LPC54606J256";
-		flash_size = 0x40000;
-		break;
-	case 0x7f954607U:
-		t->driver = "LPC54607J256";
-		flash_size = 0x40000;
-		break;
-	case 0x7f954616U:
-		t->driver = "LPC54616J256";
-		flash_size = 0x40000;
-		break;
-	case 0xfff54605U:
-		t->driver = "LPC54605J512";
-		flash_size = 0x80000;
-		break;
-	case 0xfff54606U:
-		t->driver = "LPC54606J512";
-		flash_size = 0x80000;
-		break;
-	case 0xfff54607U:
-		t->driver = "LPC54607J512";
-		flash_size = 0x80000;
-		break;
-	case 0xfff54608U:
-		t->driver = "LPC54608J512";
-		flash_size = 0x80000;
-		break;
-	case 0xfff54616U:
-		t->driver = "LPC54616J512";
-		flash_size = 0x80000;
-		break;
-	case 0xfff54618U:
-		t->driver = "LPC54618J512";
-		flash_size = 0x80000;
-		break;
-	case 0xfff54628U:
-		t->driver = "LPC54628J512";
-		flash_size = 0x80000;
-		break;
-	default:
+#ifdef ENABLE_DEBUG
```

Comment:
because `chipid` is used anyway, this does not need to be wrapped in any macro logic - the DEBUG_INFO macro will take care of this for you

Reply:
Dropped the surrounding two lines, thanks.